File: app/app/schemas/stats.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class OverviewStats(BaseModel):
    total_games: int
    total_sessions: int
    total_players: int
    recent_sessions: int
    most_recent_session_date: datetime | None


class PlayerStats(BaseModel):
    player_id: int
    player_name: str
    sessions_played: int
    wins: int
    win_rate: float


class GameStats(BaseModel):
    game_id: int
    game_name: str
    times_played: int
    unique_players: int
    last_played: datetime | None


class ActivityMonth(BaseModel):
    month: str
    session_count: int

File: web/src/types/stats.ts
export interface OverviewStats {
  total_games: number;
  total_sessions: number;
  total_players: number;
  recent_sessions: number;
  most_recent_session_date: string | null;
}

export interface PlayerStats {
  player_id: number;
  player_name: string;
  sessions_played: number;
  wins: number;
  win_rate: number;
}

export interface GameStats {
  game_id: number;
  game_name: string;
  times_played: number;
  unique_players: number;
  last_played: string | null;
}

export interface ActivityMonth {
  month: string;
  session_count: number;
}

export interface PlayerGameBreakdown {
  game_id: number;
  game_name: string;
  times_played: number;
  wins: number;
  win_rate: number;
  avg_score: number | null;
  best_score: number | null;
  last_played: string | null;
}

export interface PlayerRecentSession {
  session_id: number;
  game_id: number;
  game_name: string;
  played_at: string;
  total_score: number | null;
  winner: boolean;
}

export interface PlayerProfileStats {
  player_id: number;
  player_name: string;
  created_at: string;
  sessions_played: number;
  wins: number;
  win_rate: number;
  favorite_game: string | null;
  games: PlayerGameBreakdown[];
  recent_sessions: PlayerRecentSession[];
  activity: ActivityMonth[];
}
